src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Unit.Tests/TargetSignatureTests.cs

```diff
@@ -176,24 +178,105 @@ partial interface INativeAPI : IUnmanagedInterfaceType
+            string source = $$"""
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+                using System.Runtime.InteropServices.Marshalling;
+
+                [GeneratedComInterface]
+                [Guid("0A617667-4961-4F90-B74F-6DC368E98179")]
+                partial interface IComInterface
+                {
+                    void Method();
+                }
+                """;
+
+            await VerifyComInterfaceGeneratorAsync(source, "IComInterface", "Method", (newComp, signature) =>
+            {
+                Assert.Equal(SpecialType.System_Int32, signature.ReturnType.SpecialType);
+            });
+        }
+
+        [Fact]
+        public async Task ComInterfaceMethodFunctionPointerReturnTypeChangedToOutParameter()
+        {
+            string source = $$"""
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+                using System.Runtime.InteropServices.Marshalling;
+
+                [GeneratedComInterface]
+                [Guid("0A617667-4961-4F90-B74F-6DC368E98179")]
+                partial interface IComInterface
+                {
+                    long Method();
+                }
+                """;
+
+            await VerifyComInterfaceGeneratorAsync(source, "IComInterface", "Method", (newComp, signature) =>
+            {
+                Assert.Equal(SpecialType.System_Int32, signature.ReturnType.SpecialType);
+                Assert.Equal(2, signature.Parameters.Length);
+                Assert.Equal(newComp.CreatePointerTypeSymbol(newComp.GetSpecialType(SpecialType.System_Void)), signature.Parameters[0].Type, SymbolEqualityComparer.Default);
+                Assert.Equal(newComp.CreatePointerTypeSymbol(newComp.GetSpecialType(SpecialType.System_Int64)), signature.Parameters[^1].Type, SymbolEqualityComparer.Default);
+            });
+        }
+
+        [Fact]
+        public async Task ComInterfaceMethodPreserveSigFunctionPointerReturnTypePreserved()
+        {
+            string source = $$"""
+                using System.Runtime.CompilerServices;
+                using System.Runtime.InteropServices;
+                using System.Runtime.InteropServices.Marshalling;
+
+                [GeneratedComInterface]
+                [Guid("0A617667-4961-4F90-B74F-6DC368E98179")]
+                partial interface IComInterface
+                {
+                    [PreserveSig]
+                    long Method();
+                }
+                """;
+
+            await VerifyComInterfaceGeneratorAsync(source, "IComInterface", "Method", (newComp, signature) =>
+            {
+                Assert.Equal(SpecialType.System_Int64, signature.ReturnType.SpecialType);
+                Assert.Equal(newComp.CreatePointerTypeSymbol(newComp.GetSpecialType(SpecialType.System_Void)), Assert.Single(signature.Parameters).Type, SymbolEqualityComparer.Default);
```

Comment:
Assert the signature's parameters length too?

Reply:
Assert.Single asserts the `Parameters` collection length.